Payload/ProductPayload.php

```diff
@@ -38,6 +39,61 @@ public function getCode(): string
         return $this->code;
     }
 
+    public function isEnabled(): bool
+    {
+        return $this->payload->getBoolValue('enabled');
+    }
+
+    public function getMainTaxonId(): int
+    {
+        return $this->payload->getIntValue('mainTaxonId');
+    }
+
+    /**
+     * @return int[]
+     */
+    public function getTaxonIds(): array
+    {
+        return array_map(function (array $taxon) {
+            return $taxon['taxonId'];
+        }, $this->payload->getArrayValue('productTaxons'));
+    }
+
+    /**
+     * @return ProductTranslationPayload[]
+     */
+    public function getTranslations(): array
+    {
+        $translations = [];
+        foreach ($this->payload->getArrayValue('translations') as $translationPayload) {
+            $translation = new ProductTranslationPayload($translationPayload);
+            $translations[$translation->getLocale()] = $translation;
+        }
+
+        return $translations;
+    }
+
+    /**
+     * @return ProductVariantPayload[]
+     */
+    public function getVariants(): array
+    {
+        $variants = [];
+        foreach ($this->payload->getArrayValue('variants') as $variantPayload) {
+            $variants[] = new ProductVariantPayload($variantPayload['code'], $variantPayload);
+        }
+
+        return $variants;
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function getImages(): array
```

Comment:
would add a test for the sake of this for completion 🙂

Reply:
?? a test which tests the exception ?? i am not sure if that makes sense 🙈

Reply:
it would make sense because it forces you to think about the testcase when implementing the method 🙂